### PR TITLE
Tweaks

### PR DESCRIPTION
**Pull request by smolkaj**
- Add .bazel suffix to WORKSPACE/BUILD files to avoid confusing people unfamiliar with Bazel.
- Make p4runtime dependency more robust:
  - Use same version as cmake, i.e. the submodule loaded as `control-plane/p4runtime`.
  - Don't use custom BUILD file.
     - This required first fixing the upstream Bazel rules... (see https://github.com/p4lang/p4-constraints/pull/27)
     - ... as well as updating the submodule to point to the latest commit.
- Make googletest dependency more robust: 
  - Use same version as cmake, i.e. the submodule loaded as `test/frameworks/gtest`.
  - Don't use custom BUILD file.
    - This required updating the submodule to a newer upstream version of gtest.
- Factor repository setup into `p4c_deps` rule to make p4c easy to use by other Bazel projects.

### CODE REVIEW

**Comment by smolkaj** [comment]
@EileenFeng @stefanheule could you take a look if this looks reasonable?

File: BUILD.bazel [hunk 1]
```diff
@@ -1,14 +1,11 @@
...
 package(
     default_visibility = ["//visibility:public"],
+    license = "notice",  # Apache v2
 )
 
-PACKAGE_NAME = "com_github_p4lang_p4c"
-
 P4C_BUILD_DEFAULT_COPTS = [
     "-DCONFIG_PKGDATADIR=\\\"external/p4lang_p4c\\\"",
     # This is a bit of a hack, but will work if the binary is executed by Bazel
```
Comment (stefanheule) [inline BUILD.bazel]: I don't really understand this, is this hack necessary? (I know this is not something you changed, just curious).

File: .gitignore [hunk 2]
```diff
@@ -37,3 +37,6 @@ build/
 
 # macOS files
```
Comment (stefanheule) [inline .gitignore:38]: Are we still working on a p4_library bazel function?

**Comment by stefanheule** [comment]
Overall this looks great though, I'm super excited to get p4c opensource bazel rules.

File: .gitignore (hunk 2 above)
Comment (smolkaj) [reply .gitignore:38]: Yes.

File: BUILD.bazel (hunk 1 above)
Comment (smolkaj) [reply BUILD.bazel]: I'd like to understand that as well. Not sure yet.

Comment (smolkaj) [reply BUILD.bazel]: Let's leave this for another time.

File: .gitignore (hunk 2 above)
Comment (smolkaj) [reply .gitignore:38]: Done.

**Pull request merged**